docs/clustering.md

```diff
@@ -0,0 +1,51 @@
+# Clustering
+
+The Event Gateway instances use a strongly consistent, subscribable DB (initially [etcd](https://coreos.com/etcd),
+with support for Consul, and Zookeeper planned) to store and broadcast configuration. The instances locally
+cache configuration used to drive low-latency event routing. The instance local cache is built asynchronously based on
+events from backing DB.
+
+The Event Gateway is a horizontally scalable system. It can be scaled by adding instances to the cluster. A cluster is
```

Comment:
I feel this paragraph that introduces the notion of clustering should come first.